hackathon_site/hardware/tests.py

```diff
@@ -1,3 +1,71 @@
 from django.test import TestCase
 
-# Create your tests here.
+from hardware.models import Hardware, Category
+from hardware.serializers import HardwareSerializer, CategorySerializer
+
+
+class HardwareSerializerTestCase(TestCase):
```

Comment:
There should be extra tests here for the different cases that `get_quantity_remaining()` can have. 

- With and without any OrderItems
- With and without any returned items
- With and without anything in the cart

You can combine those together into just a few tests, definitely don't need 8 cases.